### ISSUE TITLE
Task: obtain the peak price tracking paper and record its update rule, so the second algorithm set can build PeakPriceTracking

### ISSUE THREAD

**Issue by dcelisgarza** [dcelisgarza/PortfolioOptimisers.jl#1158]
## Question

**Obtain the peak price tracking paper and record its update rule**, so that the row can be built in the second algorithm set of [ADR 0156](https://github.com/dcelisgarza/PortfolioOptimisers.jl/blob/dev/docs/adr/0156-every-online-selection-algorithm-ships-as-a-closed-form-rule-an-expert-mixture-or-a-follow-the-leader-over-a-selected-sample.md).

The literature ledger ([Research: the ledger of online selection algorithms](https://github.com/dcelisgarza/PortfolioOptimisers.jl/issues/1149), §2.3 and §7) could not read the paper: Lai, Dai, Ren and Huang (2018), *A peak price tracking-based learning system for portfolio selection*, IEEE Transactions on Neural Networks and Learning Systems 29(7), is paywalled and bot-blocked. The row rests on the abstract and a one-line restatement in a later paper: the forecast is the recent peak over the last price, `max_{0<=k<w} p_{t-k} / p_t`, followed by a linear "transform function" step towards the forecast with rate `eta`, projected. **The closed form of that step is unverified**, and `PeakPriceTracking` cannot be built from a guess.

### Checklist (HITL)

1. Fetch the paper through institutional access.
2. Record, on this ticket: the update rule as the paper writes it (the objective, its closed-form or iterative solution, and the projection); the defaults (`w`, `eta`, and any other hyperparameter); and whether the step is a Euclidean projection, as the abstract suggests, or an inner programme.
3. Note any difference from the later restatement (`eta = 100`, `w = 5`).

### Resolution

The answer is the update rule and its defaults, in the ledger's notation, so the set-2 build ticket can cite it by author and year. Nothing on this ticket names an implementation.

**Comment by dcelisgarza** [comment]
## Resolution

The paper was fetched (Lai, Dai, Ren and Huang, 2018, IEEE TNNLS 29(7), DOI 10.1109/TNNLS.2017.2705658) and read in full. **The step is a closed form — Algorithm 1 of the paper — built from two Euclidean projections. There is no inner programme.** What follows is the rule in the ledger's notation, with the paper's equation numbers.

### 1. The update rule as the paper writes it

**Forecast** (their Eq. 8; §III-A). Over the window of the last `w` price levels, including the current one,

```
phat_{t+1,i} = max_{0 <= k <= w-1} p_{t-k,i},        xhat_{t+1} = phat_{t+1} ./ p_t
```

In price relatives, `xhat_{t+1,i} = max(1, max_{1<=k<=w-1} 1 / prod_{j=0}^{k-1} x_{t-j,i})`, so the window costs `w - 1` price relatives, and `xhat >= 1` elementwise (the peak is never below the last price). The ledger's row had this right.

**Objective** (their Eq. 9; §III-B):

```
bhat_{t+1} = argmax_{b in Delta_d} <b, xhat_{t+1}>    s.t.  ||b - bhat_t||_2 <= eps,   eps > 0
```

**Solution** (§III-C, Eqs. 13–22, Propositions 1–2). The non-negativity constraint is dropped, the budget constraint is kept, and the step is solved in the increment `c_{t+1} = b_{t+1} - bhat_t`:

```
xhat_perp = (I - (1/d) 1 1') xhat_{t+1}                        # Eq. 18: the centred forecast, in the subspace 1'c = 0
chat      = eps * xhat_perp / ||xhat_perp||_2   if xhat_perp != 0    # Eq. 20: the maximiser of <c, xhat> over the eps-ball in that subspace (Cauchy–Schwarz)
chat      = 0                                   otherwise            # hold
btilde    = bhat_t + chat                                            # Eq. 21, "learning"
bhat_{t+1} = Proj_Delta(btilde) = argmin_{b in Delta_d} ||b - btilde||_2^2   # Eq. 21, "projection", Duchi et al. (2008)
```

The paper's "transform function" is therefore `f(xhat_{t+1}) = Proj_eps((I - (1/d) 1 1') xhat_{t+1})` (their Eq. 22), the Euclidean projection of the centred forecast onto the `eps`-ball. It is a function of the forecast alone — the current portfolio enters only as the point the increment is added to.

Two facts the docstring should state, because they are the paper's own:

- **The closed form is the exact solution of the relaxed problem (their Eq. 17: ball ∩ budget hyperplane), and the simplex projection restores non-negativity afterwards.** The composite is Algorithm 1, not the exact maximiser of Eq. 9. Same drop-then-project pattern as OLMAR.
- **`eps` is a step length, not a rate.** Whenever `xhat_perp != 0`, `||btilde - bhat_t||_2 = eps` exactly, whatever the size of the forecast. The forecast fixes only the direction.

**Hold case.** `xhat_perp = 0` iff every asset has the same peak-over-last ratio — in particular when every asset sits at its own window peak (`xhat = 1`). Then `bhat_{t+1} = Proj_Delta(bhat_t) = bhat_t`.

### 2. Defaults (§III-D, §IV)

- `w = 5`, "a common setting" shared with OLMAR, RMR, CORN and Anticor in their experiments.
- `eps = 100`. The sensitivity study (their Fig. 4) sweeps `eps in {5, 10, 50, 100, 500, 1000}` at `w = 5` on five data sets: cumulative wealth is flat for `eps >= 50` on all five, and `eps = 5` is markedly worse on three; `eps = 100` is chosen as "a little better than 50" (it wins on the Toronto data set).
- Initial portfolio `b_1 = (1/d) 1`; the learning process starts at the `w`-th period.
- Their comparison settings, which corroborate the ledger's other rows: OLMAR `alpha = 0.5, eps = 10`; RMR `w = 5, eps = 5`; CORN `w = 5, P = 1, rho = 0.1`; Anticor `w = 5`.

**The default saturates the step.** With weights in the simplex and a unit direction `d = xhat_perp / ||xhat_perp||`, the simplex projection of `bhat_t + eps * d` is the one-hot on `i* = argmax_i d_i` whenever `eps * (d_{i*} - d_j) >= 2` for every other `j` (sufficient; the exact condition is `eps * (d_{i*} - d_j) >= 1 + b_j - b_{i*}`). At `eps = 100` that is a gap of `0.02` in the unit direction, so **the paper's default is, in practice, "all wealth on the asset with the highest peak-over-last ratio"**, and the ball only bites on near-ties. The flat curve for `eps >= 50` in their Fig. 4 is this saturation. The paper's own worked example (their Fig. 3, six assets, `xhat_perp = (-0.042, 0.069, 0.045, -0.032, -0.004, -0.036)`) gives a top-two gap of `0.23` in the unit direction — one-hot on asset 2. A parity test at the default must expect one-hot outputs; a test that exercises the ball must set `eps` well below 1.

### 3. Euclidean projection or inner programme?

**Euclidean, twice, closed form.** The centred-ball step is scalar (a normalisation), and the final projection is the Euclidean projection onto the simplex. In the vocabulary of [ADR 0159](https://github.com/dcelisgarza/PortfolioOptimisers.jl/blob/dev/docs/adr/0159-a-constrained-online-update-projects-onto-an-allocation-set-in-the-rules-own-geometry-and-the-default-set-needs-no-solver.md) the rule's geometry is Euclidean, the centring and the ball are the rule's own learning step, and the last line of Algorithm 1 is the projection onto the allocation set. The paper's complexity claim is `O(d^2 n)` from the centring matrix product, which is `O(d)` when written as `xhat - mean(xhat)`.

### 4. Differences from the later restatement (`eta = 100`, `w = 5`)

- The symbol is **`eps`**, not `eta`; the numbers `w = 5`, `eps = 100` agree.
- "A linear transform-function step towards the forecast with rate `eta`" is wrong in kind: the step has **fixed Euclidean length `eps`** along the **centred** forecast, and it is followed by the simplex projection. There is no rate, no difference `xhat - b`, and no dependence on the forecast's magnitude.
- The ledger's §7 "unverified" flag on this row is cleared; the row's forecast column was correct, and its update column reads as in §1 above.

### 5. Beyond the row

The paper closes (§V, Eqs. 28–29) with an unbuilt extension: a nonlinear transform `f(b, xhat) = A phi(b) + psi(xhat)` over Gaussian radial basis functions, left "open". No ledger row asks for it; it is not part of set 2.

The paper's citation for the set-2 build ticket: Lai, Dai, Ren and Huang (2018), *A peak price tracking-based learning system for portfolio selection*, IEEE Transactions on Neural Networks and Learning Systems 29(7) — window peak forecast (their Eq. 8), Algorithm 1, `w = 5`, `eps = 100`. Simplex projection: Duchi, Shalev-Shwartz, Singer and Chandra (2008), ICML.

**Comment by dcelisgarza** [comment]
Resolved: the paper was fetched and read; the update rule, its closed form, the two Euclidean projections and the defaults (w = 5, eps = 100, a step length rather than a rate, saturating to one-hot at the default) are recorded in the resolution comment above. Nothing on this ticket names an implementation.
